ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -87,26 +98,89 @@ static int azure_file_read(WT_FILE_HANDLE *, WT_SESSION *, wt_off_t, size_t, voi
   __attribute__((__unused__));
 static int azure_file_size(WT_FILE_HANDLE *, WT_SESSION *, wt_off_t *) __attribute__((__unused__));
 
+// Return a customised file system to access the Azure storage source.
 static int
 azure_customize_file_system(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session,
-  const char *bucket_name, const char *auth_token, const char *config,
-  WT_FILE_SYSTEM **file_systemp)
+  const char *bucket, const char *auth_token, const char *config, WT_FILE_SYSTEM **file_system)
 {
-    WT_UNUSED(storage_source);
-    WT_UNUSED(session);
-    WT_UNUSED(bucket_name);
-    WT_UNUSED(auth_token);
-    WT_UNUSED(config);
-    WT_UNUSED(file_systemp);
-
-    return 0;
+    if (bucket == nullptr || std::string(bucket).length() == 0) {
+        std::cerr << "azure_customize_file_system: Bucket not specified." << std::endl;
+        return EINVAL;
+    }
+
+    // Get any prefix to be used for the object keys.
+    WT_CONFIG_ITEM obj_prefix_config;
+    std::string obj_prefix;
+
+    // Get the value of the config key from the string
+    azure_store *azure_storage = reinterpret_cast<azure_store *>(storage_source);
+    int ret;
+    if ((ret = azure_storage->wt_api->config_get_string(
+           azure_storage->wt_api, session, config, "prefix", &obj_prefix_config)) == 0)
+        obj_prefix = std::string(obj_prefix_config.str, obj_prefix_config.len);
+    else if (ret != WT_NOTFOUND) {
+        std::cerr << "azure_customize_file_system: error parsing config for object prefix."
+                  << std::endl;
+        return ret;
+    }
+
+    // Fetch the native WT file system.
+    WT_FILE_SYSTEM *wt_file_system;
+    if ((ret = azure_storage->wt_api->file_system_get(
+           azure_storage->wt_api, session, &wt_file_system)) != 0)
+        return ret;
+
+    // Get a copy of the home directory.
+    const std::string home_dir = session->connection->get_home(session->connection);
```

Comment:
I'd move this further down, i.e. `azure_fs->home_dir = std::string(session->connection->get_home(...))`

Reply:
wrapping `session->connection->get_home(...)`  with `std::string` limited it's lifetime to just that line (destructor called when exiting scope), and gave failed connection errors when python testing. Moved it down and changed to:
`azure_fs->home_dir = session->connection->get_home(session->connection)`